src/main/java/de/samply/share/broker/rest/Searchbroker.java

```diff
@@ -1271,8 +1274,63 @@ public Response setSiteByName(@HeaderParam(HttpHeaders.AUTHORIZATION) String aut
     }
   }
 
+  /**
+   * Save structured query.
+   * @param queryContainer the structuredQuery and the targets
+   * @return response if the query has been created
+   */
+  @POST
+  @Path("/executeSQ")
+  @Produces(MediaType.TEXT_PLAIN)
+  @Consumes({MediaType.APPLICATION_JSON, MediaType.TEXT_PLAIN})
+  @APIResponses({
+      @APIResponse(
+          responseCode = "200",
+          description = "ok",
+          content = @Content(
+              mediaType = MediaType.APPLICATION_JSON,
+              schema = @Schema(implementation = String.class))),
+      @APIResponse(responseCode = "500", description = "Internal Server Error")
+  })
+  @Operation(summary = "Save query in searchbroker database")
+  public Response executeSQ(@HeaderParam(HttpHeaders.AUTHORIZATION) String auth,
+      @Parameter(
+          name = "structuredQuery",
+          description = "Query as a JSON object",
+          schema = @Schema(implementation = QueryContainer.class))
+          QueryContainer<String> queryContainer) {
+    try {
+      if (auth == null || !checkBasicAuth(auth, "abide.username", "abide.password")) {
+        return Response.status(Response.Status.UNAUTHORIZED).build();
+      }
+      return Response.ok(
+          SearchController.releaseQuery(queryContainer.getQuery(),
+              queryContainer.getTarget(), queryContainer.getQueryName(), true)).build();
+    } catch (Exception e) {
+      LOGGER.warn(e.getMessage());
+      return Response.status(Response.Status.INTERNAL_SERVER_ERROR).build();
+    }
+  }
 
 
+  /**
+   * Save structured query.
+   *
+   * @return 200 or 500 code
```

Comment:
First a Response is returned and not a code. Second, the code is 204 and not 200. Third, it's better to explain in which cases which response is returned and why.